rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -461,13 +466,8 @@ class ReaderInfo : public ReaderListener
     mapmutex.lock();
     auto it = topicNtypes.find(fqdn);
     if (change->kind == ALIVE) {
-      if (
-        it == topicNtypes.end() ||
-        it->second.find(proxyData.typeName()) == it->second.end())
-      {
-        topicNtypes[fqdn].insert(proxyData.typeName());
-        trigger = true;
-      }
+      topicNtypes[fqdn].insert(proxyData.typeName());
+      trigger = true;
```

Comment:
With `it` being removed from the `if` block the declaration of `auto it = ...` can be moved into the `else` block.

Same below.